addons/sale/controllers/product_configurator.py

```diff
@@ -32,9 +36,15 @@ def optional_product_items(self, product_id, pricelist_id, **kw):
 
     @http.route(['/product_configurator/get_combination_info'], type='json', auth="user", methods=['POST'])
     def get_combination_info(self, product_template_id, product_id, combination, add_qty, pricelist_id, **kw):
-        return self._get_combination_info(product_template_id, product_id, combination, add_qty, self._get_pricelist(pricelist_id))
+        combination = request.env['product.template.attribute.value'].browse(combination)
+        return request.env['product.template'].browse(int(product_template_id))._get_combination_info(combination, int(product_id or 0), int(add_qty or 1), self._get_pricelist(pricelist_id))
+
+    @http.route(['/product_configurator/create_product_variant'], type='json', auth="user", methods=['POST'])
```

Comment:
@seb-odoo This should be auth="public" to reflect the changes in https://github.com/odoo/odoo/pull/27726

Reply:
I kept it 'user' in sale but override it to 'public' when website_sale is installed.